src/gen/conditions.rs

```diff
@@ -0,0 +1,1568 @@
+                range_cache,
+                flags,
+            )?;
+            Ok(Condition::ReserveFee(fee))
+        }
+        CREATE_COIN_ANNOUNCEMENT => {
+            let msg = sanitize_announce_msg(a, first(a, c)?, ErrorCode::InvalidCoinAnnouncement)?;
+            Ok(Condition::CreateCoinAnnouncement(msg))
+        }
+        ASSERT_COIN_ANNOUNCEMENT => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertCoinAnnouncementFailed)?;
+            Ok(Condition::AssertCoinAnnouncement(id))
+        }
+        CREATE_PUZZLE_ANNOUNCEMENT => {
+            let msg = sanitize_announce_msg(a, first(a, c)?, ErrorCode::InvalidPuzzleAnnouncement)?;
+            Ok(Condition::CreatePuzzleAnnouncement(msg))
+        }
+        ASSERT_PUZZLE_ANNOUNCEMENT => {
+            let id = sanitize_hash(
+                a,
+                first(a, c)?,
+                32,
+                ErrorCode::AssertPuzzleAnnouncementFailed,
+            )?;
+            Ok(Condition::AssertPuzzleAnnouncement(id))
+        }
+        ASSERT_MY_COIN_ID => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyCoinIdFailed)?;
+            Ok(Condition::AssertMyCoinId(id))
+        }
+        ASSERT_MY_PARENT_ID => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyParentIdFailed)?;
+            Ok(Condition::AssertMyParentId(id))
+        }
+        ASSERT_MY_PUZZLEHASH => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyPuzzlehashFailed)?;
+            Ok(Condition::AssertMyPuzzlehash(id))
+        }
+        ASSERT_MY_AMOUNT => {
+            let amount = parse_amount(
+                a,
+                first(a, c)?,
+                ErrorCode::AssertMyAmountFailed,
+                range_cache,
+                flags,
+            )?;
+            Ok(Condition::AssertMyAmount(amount))
+        }
+        ASSERT_SECONDS_RELATIVE => Ok(Condition::AssertSecondsRelative(parse_seconds(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertSecondsRelative,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_SECONDS_ABSOLUTE => Ok(Condition::AssertSecondsAbsolute(parse_seconds(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertSecondsAbsolute,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_HEIGHT_RELATIVE => Ok(Condition::AssertHeightRelative(parse_height(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertHeightRelative,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_HEIGHT_ABSOLUTE => Ok(Condition::AssertHeightAbsolute(parse_height(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertHeightAbsolute,
+            range_cache,
+            flags,
+        )?)),
+        _ => Err(ValidationErr(c, ErrorCode::InvalidConditionOpcode)),
+    }
+}
+
+#[derive(Debug)]
+pub struct Npc {
```

Comment:
`NPC` was always a terrible name, and since this isn't quite exactly the same, this gives us the opportunity to rename it. Maybe `SpendConditionsSummary`?

Reply:
fixed